model/user.js

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable comma-dangle */
+const usersDatabase = require("./databaseLogic/usersDatabase");
+
+class User {
+  constructor(usersDatabaseClass = usersDatabase) {
+    this.usersDatabaseClass = usersDatabaseClass;
+  }
+
+  async addUser(username, password, email) {
+    const newUser = await this.usersDatabaseClass.addUser(
+      username,
+      password,
+      // eslint-disable-next-line comma-dangle
+      email
+    );
+    return {
+      id: newUser[0].id,
+      username: newUser[0].username,
+      email: newUser[0].email,
+    };
+  }
+
+  async authenticate(username, password) {
+    const result = await this.usersDatabaseClass.findByUsername(username);
+    if (result.length === 0) {
+      return "incorrect username";
+    }
+    if (result[0].password !== password) {
+      return "incorrect password";
+    }
+    return {
+      id: result[0].id,
+      username: result[0].username,
+      email: result[0].email,
+    };
+  }
+
+  async isUsernameTaken(username) {
```

Comment:
As mentioned in the controller - these methods should be called from within the addUser method above, which would then throw the appropriate errors back to the controller if the input is not valid.